selfservice/strategy/link/strategy_recovery.go

```diff
@@ -271,6 +272,7 @@ func (s *Strategy) recoveryIssueSession(w http.ResponseWriter, r *http.Request,
 	if err != nil {
 		return s.retryRecoveryFlowWithError(w, r, flow.TypeBrowser, err)
 	}
+	sf.RequestURL = f.RequestURL
```

Comment:
This would incorrectly use the original recovery flow as the request URL. But we only need the `return_to` query parameter - I would suggest we use only that part!